core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,625 @@
+         2. Click on "Azure Active Directory" once in the "Admin Centers"
+         3. Applications --> Enterprise applications --> Security --> Conditional Access 
+         4. Make sure there is a Policy with following conditions: 
+         
+           * Grant --> Grant access (Require multifactor authentication)
+           * Users --> Users and groups --> Include --> Select users and groups --> Directory roles --> at minimum select following rules (Billing admin, Conditional Access admin, Exchange admin, Global admin, Helpdesk admin, Security admin, SharePoint admin, and User admin)
+           * Cloud apps or actions --> All cloud apps 
+         5. Make sure that the policy is Enabled  
+    query: |
+      msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'AdminMFAV2' && _['score'] == 10 )
+  - uid: mondoo-m365-ensure-that-between-two-and-four-global-admins-are-designated
+    title: Ensure that at least three global admins are designated and not more than four
+    severity: 80
+    docs:
+      desc: |
+        This control checks that there are enough Global Admins in a single tenant.
+      audit: |
+        __cnquery run__
+
+        To audit Microsoft MS365 with `cnquery run`:
+
+        Run the following query:
+
+          ```mql
+          cnquery run ms365 -c "msgraph.beta.rolemanagement.roleDefinitions.where(displayName == "Global Administrator").all(assignments.length > 1 && assignments.length <= 4)" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.rolemanagement.roleDefinitions.where(displayName == "Global Administrator").all(assignments.length > 1 && assignments.length <= 4)
+          ```
+      remediation: |
+        ### MS365 Console
+         
+         To update via the ms365 portal:
+
+         1. Log in as Global Administrator to the ms365 portal at https://admin.microsoft.com
+         2. Click on "Azure Active Directory" once in the "Admin Centers"
+         3. Users --> Active Users  
+         4. Create a new Global Admin, if there are not enough by: 
+            * Select the User's name
+            * Manage roles --> Admin center access -> Global Administrator -> Save changes
+         5. Remove the Global Admin, if there are more than four Global Administrators
+            * Select the User
+            * Roles --> Manage roles --> De-Select the appropriate role -> Save changes  
+    query: |
+      msgraph.beta.rolemanagement.roleDefinitions.where(displayName == "Global Administrator").all(assignments.length > 1 && assignments.length <= 4)
```

Comment:
`msgraph.beta.rolemanagement` is deprecated, use the latest version

Reply:
Thanks Chris, done!